include/geos/index/strtree/TemplateSTRtree.h

```diff
@@ -306,6 +306,8 @@ class TemplateSTRtreeImpl {
     /// @{
 
     bool remove(const BoundsType& itemEnv, const ItemType& item) {
+        build();
```

Comment:
Unrelated to the actual PR, but something that I never found fully clear in the C API docs, and was reminded about when seeing this change.

The comments in this file state:

> once the tree has been built (explicitly or on the first call to `query`), items may not be added or removed.

For which reason I always wondered a bit what the use case if for `GEOSSTRtree_remove` (we also don't expose it in Shapely). But if that comment is correct, and calling `remove` also builds the tree, does that mean you can only call `remove` once? That would be strange, but then the comment is no longer up to date?

Reply:
Yeah, I think the comment is wrong. More items cannot be added, but they can be "removed" -- which really means "retained but flagged to not be returned in queries."